### PR TITLE
minor improvements for fract and trunc

### PR DESCRIPTION
**Pull request by redmoogle**
cleaner code also probably more performant

### CODE REVIEW

File: OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs [hunk 1]
```diff
@@ -709,11 +709,14 @@ static class DreamProcNativeRoot {
...
-                if(float.IsInfinity(floatnum)) {
+            if (arg.TryGetValueAsFloat(out var floatNum)) {
+                if(float.IsInfinity(floatNum)) {
                     return new DreamValue(0);
                 }
-                return new DreamValue(floatnum - MathF.Truncate(floatnum));
+                if(float.IsNaN(floatNum)) {
+                    return arg;
+                }
+                return new DreamValue(floatNum - (int)floatNum);
```
Comment (wixoaGit) [inline OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs:719]: Why should `(int)floatNum` be preferred over `MathF.Truncate(floatNum)`?

File: OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs [hunk 2]
```diff
@@ -2720,8 +2723,11 @@ public static DreamValue NativeProc_sorttext(NativeProc.State state)
         [DreamProcParameter("n", Type = DreamValueType.Float)]
         public static DreamValue NativeProc_trunc(NativeProc.State state) {
             DreamValue arg = state.GetArgument(0, "n");
-            if (arg.TryGetValueAsFloat(out float floatnum)) {
-                return new DreamValue(MathF.Truncate(floatnum));
+            if (arg.TryGetValueAsFloat(out var floatNum)) {
+                if (!float.IsFinite(floatNum)) {
+                    return arg;
+                }
+                return new DreamValue((int)floatNum);
```
Comment (wixoaGit) [inline OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs:2730]: Does this not just give the same values as `MathF.Truncate()`?

**Review by wixoaGit — CHANGES_REQUESTED** [review]

File: OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs (hunk 1 above)
Comment (redmoogle) [reply OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs:719]: they do roughly the same things do they not; subtraction shouldnt be a particularity expensive op. Although im not 100% sure on intcasting

File: OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs (hunk 2 above)
Comment (redmoogle) [reply OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs:2730]: Yes, but imo in this case its cleaner since its just turning it into a integer

Comment (wixoaGit) [reply OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs:2730]: I don't see how this is cleaner and preferable over a one-line call to the `MathF` equivalent of the proc.

**Comment by wixoaGit** [comment]
This doesn't affect the behavior or performance of these procs, nor is it better looking code in my opinion.

**Pull request closed without merging**